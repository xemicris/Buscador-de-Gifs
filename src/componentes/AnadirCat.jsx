import { useState } from "react"

export const AnadirCat = ({onNuevaCategoria, onNuevoLimite}) => {

  const [inputValor, setInputValor] = useState('');
  const [limite, setLimite] = useState(5);

  //desestructura el evento
  const inputCambio = ({target}) =>{
    setInputValor(target.value);
  }

  const limiteCambio = ({target}) =>{
    setLimite(target.value);
    console.log(limite);
  }

  const onSubmit = (evento)=>{
    evento.preventDefault();
    const inputValorLimpio = inputValor.trim();
    if(inputValorLimpio.length <= 1) return;
    onNuevaCategoria(inputValorLimpio);
    onNuevoLimite(limite);
    //limpiarlo
    setInputValor('');
  }

  return (
    <form onSubmit={onSubmit} className="formulario">
      <input
        type="text"
        placeholder="Nombre del Gif"  
        value={inputValor} 
        onChange={inputCambio} 
        className="inputs input-entrada"
      />
       <input
          type="number"
          value={limite}
          onChange={limiteCambio}
          className="inputs input-limite"
          max={10}
        />
        <button className="boton">Enviar</button>
    </form>
  )
}
