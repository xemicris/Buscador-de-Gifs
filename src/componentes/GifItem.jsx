import PropTypes from 'prop-types'

//Componente que muestra las tarjetas
export const GifItem = ({titulo, urlImagen, id}) => {

  return (
    <div className="carta">
        <img className="imagen" src={urlImagen} alt={titulo} />
        <p className="titulo-categoria">{titulo}</p>
    </div>
  )
}

GifItem.propTypes={
  titulo: PropTypes.string.isRequired,
  urlImagen: PropTypes.string.isRequired
}
