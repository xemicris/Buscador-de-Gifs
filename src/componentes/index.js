

export * from './AnadirCat';
export * from './GifGrid';
export * from './GifItem';