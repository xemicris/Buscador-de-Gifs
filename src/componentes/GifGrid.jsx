import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

//Componente que crea la estructura de cada categoría
export const GifGrid = ({categoria, limite}) => {

  const {imagenes, estaCargando} = useFetchGifs(categoria, limite); 

  return (
    <>
        <p className="categoria">{categoria}</p>
        {
          estaCargando ? (<p className="cargando">Cargando...</p>) : null
        }
        <div className="carta-grid">
          {/* En el paréntesis tras la flecha va el return implícito */}
          {imagenes.map(imagen =>(
              <GifItem 
                key={imagen.id}
                {...imagen}
              />
            )
          )}
        </div>

    </>
  )
}
