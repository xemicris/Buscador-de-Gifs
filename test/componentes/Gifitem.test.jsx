import { GifItem } from "../../src/componentes/GifItem"
import {render, screen} from '@testing-library/react';


describe('Pruebas en <GifItem. />', () => { 
    
    const tit = 'Saint-Seiya';
    const url = 'https://localhost/saintseiya.jpg';

    test('debe hacer match con el snapshot', () => { 

        const {container} = render(<GifItem titulo={tit} urlImagen={url} />);

        expect(container).toMatchSnapshot();
     })

     test('Debe mostrar la imagen con el URL y el ALT indicados', () => { 
        
         render(<GifItem titulo={tit} urlImagen={url} />);
        // expect(screen.getByRole('img').src).toBe( url );
        // expect(screen.getByRole('img').alt).toBe(tit);

        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(tit);

      })

      test('debe de mostrar el tÃ­tulo en el componente', () => { 
        
        render(<GifItem titulo={tit} urlImagen={url} />);
        expect(screen.getByText(tit)).toBeTruthy();
       })
 })