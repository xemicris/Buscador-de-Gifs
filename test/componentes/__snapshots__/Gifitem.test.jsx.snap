// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem. /> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="carta"
  >
    <img
      alt="Saint-Seiya"
      class="imagen"
      src="https://localhost/saintseiya.jpg"
    />
    <p
      class="titulo-categoria"
    >
      Saint-Seiya
    </p>
  </div>
</div>
`;
